### PR TITLE
Collapse public and private pod xcconfig into one

### PR DESCRIPTION
**Pull request by mrackwitz**
The public pod xcconfigs used to define namespaced xcconfig values for the former spec attribute `xcconfig`. This was originally introduced by https://github.com/CocoaPods/CocoaPods/pull/1011.
This was partially reverted with https://github.com/CocoaPods/CocoaPods/issues/1189.
With https://github.com/CocoaPods/CocoaPods/pull/3573, we changed the approach on populating build settings to the user target fundamentally, so that the public pod xcconfigs became finally completely dispensable. This PR collapses what they still did with the private pod xcconfig.

### CODE REVIEW

File: spec/unit/generator/xcconfig/pod_xcconfig_spec.rb [hunk 1]
```diff
@@ -10,8 +10,18 @@ module XCConfig
...
-            @generator = PrivatePodXCConfig.new(@pod_target, public_xcconfig)
+            @generator = PodXCConfig.new(@pod_target)
+
+            @spec.xcconfig = { 'OTHER_LDFLAGS' => '-no_compact_unwind' }
+            @spec.frameworks = ['QuartzCore']
+            @spec.weak_frameworks = ['iAd']
+            @spec.libraries = ['xml2']
+            file_accessors = [Sandbox::FileAccessor.new(fixture('banana-lib'), @spec.consumer(:ios))]
+            # vendored_framework_paths = [config.sandbox.root + 'BananaLib/BananaLib.framework']
+            # Sandbox::FileAccessor.any_instance.stubs(:vendored_frameworks).returns(vendored_framework_paths)
```
Comment (segiddins) [inline spec/unit/generator/xcconfig/pod_xcconfig_spec.rb]: why are these commented out?

**Comment by segiddins** [comment]
While you're touching this code, do you think you could add a test for https://github.com/CocoaPods/CocoaPods/issues/3906 ?

**Comment by mrackwitz** [comment]
@segiddins: That doesn't really belong into this PR, but yeah I will fix that as well.

**Comment by segiddins** [comment]
`NoMethodError: undefined method `xcconfig_private_path' for <Pod::PodTarget name=BananaLib >:Pod::PodTarget`. Also needs a changelog. Otherwise, :+1: :D.

**Comment by segiddins** [comment]
Looks like `spec/unit/installer/target_installer/pod_target_installer_spec.rb` needs an update

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,8 +6,20 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ## 0.38.2
 
+##### Enhancements
```
Comment (segiddins) [inline CHANGELOG.md]: master, not 0.38.2

**Pull request merged**